underfs/swift/src/main/java/tachyon/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -15,51 +15,370 @@
+    LOG.debug("Get object details: {}", path);
+    boolean res =  mAccount.getContainer(mContainerName).getObject(path).exists();
+    return res;
+  }
+
+  /**
+   * There is no concept of a block in Swift, however the maximum allowed size of
+   * one object is currently 4 GB.
+   * @param path to the object
+   * @return 4 GB in bytes
+   * @throws IOException this implementation will not throw this exception, but subclasses may
+   */
+  @Override
+  public long getBlockSizeByte(String path) throws IOException {
+    return Constants.GB * 4;
+  }
+
+  @Override
+  public Object getConf() {
+    LOG.warn("getConf is not supported when using SwiftDirectUnderFileSystem, returning null.");
+    return null;
+  }
+
+  @Override
+  public List<String> getFileLocations(String path) throws IOException {
+    LOG.warn("getFileLocations is not supported when using "
+        + "SwiftDirectUnderFileSystem, returning null.");
+    return null;
+  }
+
+  @Override
+  public List<String> getFileLocations(String path, long offset) throws IOException {
+    LOG.warn("getFileLocations is not supported when using "
+        + "SwiftDirectUnderFileSystem, returning null.");
+    return null;
+  }
+
+  @Override
+  public long getFileSize(String path) throws IOException {
+    StoredObject so = mAccount.getContainer(mContainerName).getObject(stripPrefixIfPresent(path));
+    return so.getContentLength();
+  }
+
+  @Override
+  public long getModificationTimeMs(String path) throws IOException {
+    StoredObject so = mAccount.getContainer(mContainerName).getObject(stripPrefixIfPresent(path));
+    return so.getLastModifiedAsDate().getTime();
+  }
+
+  @Override
+  public long getSpace(String path, SpaceType type) throws IOException {
+    return -1;
+  }
+
+  @Override
+  public boolean isFile(String path) throws IOException {
+    return exists(path);
+  }
+
+  @Override
+  public String[] list(String path) throws IOException {
+    path = path.endsWith(PATH_SEPARATOR) ? path : path + PATH_SEPARATOR;
+    return listInternal(path, false);
+  }
+
+  /* @inheritDoc
+   * @see tachyon.underfs.UnderFileSystem#mkdirs(java.lang.String, boolean)
+   * There is no notion of directories in Swift.
+   * The content of containers are objects.
+   * Object name may contain nested structure like a/b/c/d.data
+   */
+  @Override
+  public boolean mkdirs(String path, boolean createParent) throws IOException {
+    return true;
+  }
+
+  @Override
+  public InputStream open(String path) throws IOException {
+    path = stripPrefixIfPresent(path);
+    Container container = mAccount.getContainer(mContainerName);
+    StoredObject so = container.getObject(path);
+    InputStream  is = so.downloadObjectAsInputStream();
+    return is;
+  }
+
+  @Override
+  public boolean rename(String src, String dst) throws IOException {
+    if (!exists(src)) {
+      LOG.error("Unable to rename {} to {}. Source does not exists", src, dst);
+      return false;
+    }
+    copy(src, dst);
```

Comment:
Would it be possible to extend rename to handle the case of renaming a prefix (ie. directory)?